src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/Processors/GeneralizeProcess.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using EnsureThat;
+using Microsoft.Health.Fhir.Anonymizer.Core.Models;
+using Microsoft.Health.Fhir.Anonymizer.Core.Processors.Settings;
+using Microsoft.Health.Fhir.Anonymizer.Core.AnonymizerConfigurations;
+using Hl7.Fhir.ElementModel;
+using Hl7.Fhir.Model;
+using Hl7.FhirPath;
+using Newtonsoft.Json.Linq;
+using Hl7.Fhir.Model.Primitives;
+
+namespace Microsoft.Health.Fhir.Anonymizer.Core.Processors
+{
+
+    public partial class GeneralizeProcessor : IAnonymizerProcessor
+    {
+
+        private static readonly HashSet<string> s_integerValueTypeNames = new HashSet<string>
+        {
+            FHIRAllTypes.Integer.ToString(),
+            FHIRAllTypes.PositiveInt.ToString(),
+            FHIRAllTypes.UnsignedInt.ToString()
+        };
+
+        private static readonly HashSet<string> s_DateOrTimeValueTypeNames = new HashSet<string>
+        {
+            FHIRAllTypes.DateTime.ToString(),
+            FHIRAllTypes.Date.ToString(),
+            FHIRAllTypes.Time.ToString(),
+            FHIRAllTypes.Instant.ToString()
+        };
+
+        private void ValidateExpression(ElementNode node, string conditionExpression, string targetExpression)
+        {          
+            try
+            {
+                node.Predicate(conditionExpression);
+                node.Scalar(targetExpression);
+            }
+            catch (Exception ex)
+            {
+                throw new AnonymizerConfigurationErrorsException($"Invalid cases expression {conditionExpression}", ex);
+            }
+        }
+
+        private void GeneralizeIntegerNode(ElementNode node, string targetExpression)
+        {
+            if (node.Scalar(targetExpression).GetType() != typeof(Int64))
+            {
+                throw new AnonymizerConfigurationErrorsException("Invalid datatype. Expect an integer.");
+            }
+
+            int targetIntValue = Convert.ToInt32(node.Scalar(targetExpression));
```

Comment:
Could we use int instead of int32 to keep consistent with Resolute's coding style?

Reply:
seems it only has int32 option.